codalab/rest/json_api_client.py

```diff
@@ -0,0 +1,238 @@
+import base64
+from contextlib import closing
+import httplib
+import json
+import re
+import socket
+import sys
+import threading
+import time
+import urllib
+import urllib2
+import urlparse
+
+
+from codalab.client.bundle_client import BundleClient
+from codalab.common import http_error_to_exception
+
+
+def wrap_exception(message):
+    def decorator(f):
+        def wrapper(*args, **kwargs):
+            try:
+                return f(*args, **kwargs)
+            except JsonApiException as e:
+                raise JsonApiException, \
+                    JsonApiException(message + ': ' + e.message,
+                                           e.client_error), \
+                    sys.exc_info()[2]
+            except urllib2.HTTPError as e:
+                raise JsonApiException, \
+                    JsonApiException(message + ': ' +
+                                           httplib.responses[e.code] + ' - ' +
+                                           e.read(),
+                                           e.code >= 400 and e.code < 500), \
+                    sys.exc_info()[2]
+            except (urllib2.URLError, httplib.HTTPException, socket.error) as e:
+                raise JsonApiException, \
+                    JsonApiException(message + ': ' + str(e), False), \
+                    sys.exc_info()[2]
+        return wrapper
+    return decorator
+
+
+def unpack_json(raw_json):
```

Comment:
If there's anything that seems funny though please let me know lest I screw something up! :)